File: uva_406.py
import math
def prime_list(Max):
    prime=[]
    for i in range(1, Max+1):
        check_prime=True
        for j in range(2, int(math.sqrt(i))+1 ):
            if not i%j:
                check_prime=False
                break
        if check_prime:
            prime.append(i)
    return prime

while True:
    try:
        input_data= input()
    except:
        break
    Max, Mid = input_data.split()
    Max, Mid = int(Max), int(Mid)
    print("{} {}:".format(Max, Mid), end='')
    ans = prime_list(Max)
    if 2*Mid >= len(ans):
        for i in ans:
            print(" {}".format(i), end='')
        print('')
    else:
        ans_mid = int(len(ans)/2)
        start = ans_mid-Mid+(len(ans)%2)
        end = ans_mid+Mid-1
        for i in range(start, end+1):
            print(" {}".format(ans[i]), end='')
        print('')
    print('')

File: uva_406.cpp
# include <iostream>
# include <vector>
# include <cmath>
using namespace std;
vector<int> prime_list(int Max){
    vector<int> prime;
    for(int i=1; i<=Max; ++i){
        bool check_prime=true;
        for(int j=2; j<=int(sqrt(i) ) && check_prime; ++j)
            if ( !(i%j) )
                check_prime=false;
        if (check_prime)
            prime.push_back(i);
    }
    return prime;
}

int main(void){
    int Max, Mid;
    while(cin >> Max >> Mid){
        vector<int> ans=prime_list(Max);
        cout << Max << " " << Mid << ": ";
        if (2*Mid>=ans.size()){
            for(int i=0; i<ans.size(); ++i){
                cout << ans[i];
                if (i!=ans.size()-1)
                    cout << ' ';
            }
            cout << endl;
        }
        else{
            int ans_mid=ans.size()/2;
            int start = ans_mid-Mid+(ans.size()%2);
            int end = ans_mid+Mid-1;
            for(int i=start; i<=end; ++i){
                cout << ans[i];
                if (i!=end)
                    cout << ' ';
            }
            cout << endl;
        }
        cout << endl;
    }
    return 0;
}
